docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,207 @@
+
+ 1. *Initialize an empty pattern.* This can be done by either using the
+    [`init_sparsity_pattern(dh)`](@ref) function or by using a constructor directly.
+    `init_sparsity_pattern` will return a default pattern type that is compatible with the
+    DofHandler. In some cases you might require another type of pattern (for example a
+    blocked pattern) and in that case you can use the constructor directly.
+
+ 2. *Add entries to the pattern.* There are a number of functions that add entries to the
+    pattern:
+     - [`create_sparsity_pattern!`](@ref) adds entries for all couplings between the DoFs
+       within each element. These entries correspond to assembling the standard element
+       matrix and is thus almost always required.
+     - [`cross_element_coupling!`](@ref) adds entries for couplings between the DoFs in
+       neighboring elements. These entries are required when integrating along internal
+       interfaces between elements (e.g. for discontinuous Galerkin methods).
+     - [`Ferrite.add_entry!`](@ref) adds a single entry to the pattern. This can be used if
+       you need to add custom entries that are not covered by the other functions.
+
+ 3. *Condense the pattern.* This is done with the [`condense_sparsity_pattern!`](@ref)
+    function and adds entries required from constraints and boundary conditions in the
+    ConstraintHandler. Note that condensing the pattern *must* be done as the last operation
+    on the pattern, e.g. after adding all other entries.
+
+ 4. *Instantiate the matrix*. A sparse matrix can be created from the sparsity pattern using
+    [`create_matrix`](@ref), see [Instantiating the sparse matrix](@ref) below for more
+    details.
+
+!!! note "Swiss army knife"
+    The [`create_sparsity_pattern!`](@ref) function is something of a swiss army knife since
+    it can do more than just adding the basic element couplings as described above. For
+    example, when passed the correct arguments, it can also do the job of
+    `cross_element_coupling!` and `condense_sparsity_pattern!`. Refer to the API reference
+    for details.
+
+### Increasing the sparsity
+
+By default when creating a sparsity pattern it is assumed that a DoF couple with all other
+DoFs in the same element.
+
+is created such that DoFs couple with all DoFs in the same
+element. 
+
+
+
+This is the most general case and results in a dense
+matrix. However, in many cases the sparsity can be increased by reducing the number of DoFs
+that couple with each other. This can be done by using the `coupling` keyword argument in
+the `create_sparsity_pattern` function. This keyword argument takes a function that returns
+a boolean value indicating if two DoFs should couple or not. The function should take two
+arguments, the first being the element index and the second being the DoF index within the
+element. The function should return `true` if the DoFs should couple and `false` otherwise.
```

Comment:
This sentence doesn't make sense to me. 
I would have expected that `f(i, j)` should say if DoF `i` couples to DoF `j`? But here it sounds like the `cellid` is expected, but perhaps I'm just confused with array element vs finite element?

Wasn't there an option to give the field coupling via a `Matrix{Bool}` before?

Side-note: Probably giving `uncoupled::Vector{Pair{Symbol,Symbol}}` would be more user-friendly to remove couplings instead, but can be done later...

Reply:
This paragraph is Copilot hallucinations I forgot to remove haha